proto/structured.proto

```diff
@@ -104,3 +105,220 @@ message CreateTableResponse {
+  // User is the originating user. Used to lookup priority when
+  // scheduling queued operations at target node.
+  optional string user = 5 [(gogoproto.nullable) = false];
+  // UserPriority specifies priority multiple for non-transactional
+  // commands. This value should be a positive integer [1, 2^31-1).
+  // It's properly viewed as a multiple for how likely this
+  // transaction will be to prevail if a write conflict occurs.
+  // Commands with UserPriority=100 will be 100x less likely to be
+  // aborted as conflicting transactions or non-transactional commands
+  // with UserPriority=1. This value is ignored if Txn is
+  // specified. If neither this value nor Txn is specified, the value
+  // defaults to 1.
+  optional int32 user_priority = 8 [default = 1];
+  // Txn is set non-nil if a transaction is underway. To start a txn,
+  // the first request should set this field to non-nil with name and
+  // isolation level set as desired. The response will contain the
+  // fully-initialized transaction with txn ID, priority, initial
+  // timestamp, and maximum timestamp. The transaction returned in a
+  // response is feed to the next request.
+  optional Transaction txn = 9;
+  // ReadConsistency specifies the consistency for read
+  // operations. The default is CONSISTENT. This value is ignored for
+  // write operations.
+  optional ReadConsistencyType read_consistency = 10 [(gogoproto.nullable) = false];
+}
+
+// TableResponseHeader is returned with every structured API response.
+message TableResponseHeader {
+  // Error is non-nil if an error occurred.
+  optional Error error = 1;
+  // Timestamp specifies time at which read or write actually was
+  // performed. In the case of both reads and writes, if the timestamp
+  // supplied to the request was 0, the wall time of the node
+  // servicing the request will be set here. Additionally, in the case
+  // of writes, this value may be increased from the timestamp passed
+  // with the RequestHeader if the key being written was either read
+  // or written more recently.
+  //
+  // TODO(vivek): Is this only relevant in the non-transactional case?
+  optional Timestamp timestamp = 2 [(gogoproto.nullable) = false];
+  // Transaction is non-nil if the request specified a non-nil
+  // transaction. The transaction timestamp and/or priority may have
+  // been updated, depending on the outcome of the request.
+  optional Transaction txn = 3;
+}
+
+// A TableKey represents a key into a table index.
+message TableKey {
+  optional string table_name = 1;
+  // A column key component. Multiple ColumnKeys together form a key.
+  message ColumnKey {
+    optional string column_name = 1;
+    optional bytes key = 2; 
+  }
+  // The different key components from each column.
+  repeated ColumnKey columns = 2;
+}
+
+// A TableRow represents a set of column values within a table row. 
+message TableRow {
+  message Cell {
+    optional bytes column_name = 1;
```

Comment:
Does the row need the column_name? I think we can infer the column_name from the request or pull the column names out into a separate field in the responses so that we don't penalize scans which return multiple rows.